File: Data Structures & Algorithms/same-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q:
            return True
            
        if not p or not q:
            return False

        return self.dfs(p, q)

    def dfs(self, p, q) -> bool:
        if not p or not q:
            return False

        if p.val != q.val:
            return False

        if p.left or q.left:
            return self.dfs(p.left, q.left)
        if p.right or q.right:
            return self.dfs(p.right, q.right)
        
        return True

File: Data Structures & Algorithms/climbing-stairs/submission-4.cs
public class Solution {
    public int ClimbStairs(int n) {     
        return Dfs(0, n);
    }

    public int Dfs(int i, int n)
    {
        if (i >= n)
        {
            return i == n ? 1 : 0;
        }

        return Dfs(i + 1, n) + Dfs(i + 2, n);
    }
}
